galoy-client/src/client/error.rs

```diff
@@ -0,0 +1,15 @@
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum GaloyClientError {
```

Comment:
Lets start using the pattern that every crate has an error in root (instead of in sub-modules).

So move this error under `src/error`. The only exception would be for shared where there can be errors in submodules. Need to do a small refactoring to make this consistent in all sub-crates.